cudax/examples/stf/linear_algebra/cg_csr.cu

```diff
@@ -75,38 +82,40 @@ public:
   }
 
   logical_data<slice<double>> handle;
+  context ctx;
 };
 
 class scalar
 {
 public:
-  scalar()
+  scalar(const context& _ctx)
+      : ctx(_ctx)
```

Comment:
```suggestion
  scalar(context _ctx)
      : ctx(_ctx), handle(ctx.logical_data(shape_of<scalar_view<double>>()))
```

Note that the definition of member ctx needs to follow the definition of handle.